app/controllers/api/v1/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+        respond_to :json
+        def google_oauth2
+          @user = User.from_omniauth(auth)
+
+          # TODO: Return JWT token on both success and error
```

Comment:
What's the reason we will return the token in error? 🙏

Reply:
@Tuubz Ideally, when receiving a callback from Google, which means the user is already authorized, we just validate the existing user to create a new record, then we could return tokens for accessing other resources 💭